app/data_model/answer_store.py

```diff
@@ -171,6 +172,11 @@ def filter(self, group_id=None, block_id=None, answer_id=None, group_instance=No
             "answer_instance": answer_instance,
             "group_instance": group_instance,
         }
+        if location:
```

Comment:
maybe an assert/warning/exception when location  and the individual settings are both set, since its overwriting them here and the caller needs to decide what's being used.